session-replay/Sources/DatadogSessionReplay/Recorder/Utilities/CGRect+ContentFrame.swift

```diff
@@ -0,0 +1,152 @@
+            )
+        case .redraw, .center:
+            contentFrame = CGRect(
+                x: self.origin.x + (self.width - contentSize.width) / 2,
+                y: self.origin.y + (self.height - contentSize.height) / 2,
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .scaleToFill:
+            contentFrame = self
+
+        case .topLeft:
+            contentFrame = CGRect(
+                x: self.origin.x,
+                y: self.origin.y,
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .topRight:
+            contentFrame = CGRect(
+                x: self.origin.x + (self.width - contentSize.width),
+                y: self.origin.y,
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .bottomLeft:
+            contentFrame = CGRect(
+                x: self.origin.x,
+                y: self.origin.y + (self.height - contentSize.height),
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .bottomRight:
+            contentFrame = CGRect(
+                x: self.origin.x + (self.width - contentSize.width),
+                y: self.origin.y + (self.height - contentSize.height),
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .top:
+            contentFrame = CGRect(
+                x: self.origin.x + (self.width - contentSize.width) / 2,
+                y: self.origin.y,
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .bottom:
+            contentFrame = CGRect(
+                x: self.origin.x + (self.width - contentSize.width) / 2,
+                y: self.origin.y + (self.height - contentSize.height),
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .left:
+            contentFrame = CGRect(
+                x: self.origin.x,
+                y: self.origin.y + (self.height - contentSize.height) / 2,
+                width: contentSize.width,
+                height: contentSize.height
+            )
+        case .right:
+            contentFrame = CGRect(
+                x: self.origin.x + (self.width - contentSize.width),
+                y: self.origin.y + (self.height - contentSize.height) / 2,
+                width: contentSize.width,
+                height: contentSize.height
+            )
+
+        @unknown default:
+            contentFrame = self
+        }
+        return contentFrame
+    }
+}
+
+fileprivate extension CGSize {
+    func scaleAspectFillRect(for contentSize: CGSize) -> CGRect {
+        guard contentSize.width > 0 && contentSize.height > 0 else {
+            return .zero
+        }
```

Comment:
nit/ We ensure this in upstream logic (`CGRect.contentFrame(for:using:)`) and this is `fileprivate` routine, so this `guard` will never be effected. Given good tests coverage we could skip this entirely here and in (`scaleAspectFitRect(for:)`)

Reply:
Yeah - I thought about that and wanted to be on the safe side when expand this extension in the future and reuse private methods.